frb_codegen/src/generator/dart/ty.rs

```diff
@@ -16,6 +22,20 @@ pub trait TypeDartGeneratorTrait {
     fn structs(&self) -> String {
         "".to_string()
     }
+
+    fn get_context(&self) -> &TypeGeneratorContext;
+
+    fn is_type_shared(&self, ty: &IrType) -> bool {
+        self.get_context().ir_file.is_type_shared_by_safe_ident(ty)
+    }
+
+    fn get_private_prefix(&self) -> String {
+        if !self.get_context().config.shared {
+            "_".into()
+        } else {
+            "".into()
+        }
+    }
```

Comment:
nit: I am not very sure whether we should put this in trait, or as utility functions